### PR TITLE
refactor(cli): refactor cli to commander

### PR DESCRIPTION
**Pull request by samcx**
## Description
Refactor the [Next.js CLI](https://nextjs.org/docs/app/api-reference/next-cli) to use [commander](https://github.com/tj/commander.js) instead of [arg](https://github.com/vercel/arg).

## Why?
- Auto-generated, properly formatted help command + output. With `arg`, much of the help commands were manually added via a single `console.log`, causing deviations over time.
- Ergonomic, ease of adding new subcommands and rules

## Breaking Changes
- Update the experimental `next experimental-compile` and `next experimental-generate` build commands in favor of `next build --experimental-build-mode=compile/generate`

### CODE REVIEW

**Comment by samcx** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/next.js/61877?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `canary`
  * **#61748** <a href="https://app.graphite.dev/github/pr/vercel/next.js/61748?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#61877** <a href="https://app.graphite.dev/github/pr/vercel/next.js/61877?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @samcx and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 7703c0593ba2040b1ce265deca8ee43159ea3ca4

`TURBOPACK=1 pnpm test test/integration/500-page/test/gsp-gssp.test.js` (turbopack)
- gsp-gssp > development mode > does not show error with getStaticProps in pages/500 dev
- gsp-gssp > development mode > shows error with getServerSideProps in pages/500 dev

<details>
<summary>Expand output</summary>

  ● gsp-gssp › development mode › does not show error with getStaticProps in pages/500 dev

    FetchError: request to http://localhost:37767/abc failed, reason: connect ECONNREFUSED 127.0.0.1:37767

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● gsp-gssp › development mode › shows error with getServerSideProps in pages/500 dev

    dest already exists.

      at ../node_modules/.pnpm/fs-extra@9.0.0/node_modules/fs-extra/lib/move/move.js:41:31
      at ../node_modules/.pnpm/universalify@1.0.0/node_modules/universalify/index.js:21:54

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

`pnpm test test/integration/client-navigation-a11y/test/index.test.js`
- Client Navigation accessibility > <RouteAnnouncer /> > should not have the initial route announced
- Client Navigation accessibility > <RouteAnnouncer /> > has aria-live="assertive" and role="alert"
- Client Navigation accessibility > <RouteAnnouncer /> > There is a title and a h1 tag > has the innerText equal to the value of h1
- Client Navigation accessibility > <RouteAnnouncer /> > There is a title but no h1 tag > has the innerText equal to the value of document.title
- Client Navigation accessibility > <RouteAnnouncer /> > There is no title and no h1 tag > has the innerText equal to the value of the pathname
- Client Navigation accessibility > <RouteAnnouncer /> > There is no title but a h1 tag > has the innerText equal to the value of h1

<details>
<summary>Expand output</summary>

  ● Client Navigation accessibility › <RouteAnnouncer /> › should not have the initial route announced

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● Client Navigation accessibility › <RouteAnnouncer /> › has aria-live="assertive" and role="alert"

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● Client Navigation accessibility › <RouteAnnouncer /> › There is a title but no h1 tag › has the innerText equal to the value of document.title

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● Client Navigation accessibility › <RouteAnnouncer /> › There is no title but a h1 tag › has the innerText equal to the value of h1

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● Client Navigation accessibility › <RouteAnnouncer /> › There is a title and a h1 tag › has the innerText equal to the value of h1

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

  ● Client Navigation accessibility › <RouteAnnouncer /> › There is no title and no h1 tag › has the innerText equal to the value of the pathname

    FetchError: request to http://localhost:39133/page-with-h1-and-title,%20/page-with-h1,%20/page-with-title,%20/page-without-h1-or-title failed, reason: connect ECONNREFUSED 127.0.0.1:39133

      at ClientRequest.<anonymous> (../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js:1491:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| buildDuration | 12.8s | 11.9s | N/A |
| buildDurationCached | 7.2s | 5s | N/A |
| nodeModulesSize | 196 MB | 196 MB | N/A |
| nextStartRea..uration (ms) | 429ms | 588ms | ⚠️ +159ms |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| 3f784ff6-HASH.js gzip | 53.5 kB | 53.5 kB | N/A |
| 423.HASH.js gzip | 185 B | 181 B | N/A |
| 68-HASH.js gzip | 30.3 kB | 29.6 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 238 B | 239 B | N/A |
| main-HASH.js gzip | 32 kB | 31.9 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 46.9 kB | 46.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 502 B | 501 B | N/A |
| css-HASH.js gzip | 320 B | 322 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 256 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 368 B | 369 B | N/A |
| image-HASH.js gzip | 4.2 kB | 4.18 kB | N/A |
| index-HASH.js gzip | 257 B | 256 B | N/A |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 310 B | 311 B | N/A |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 306 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 106 B | 106 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 485 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 542 B | 539 B | N/A |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 528 B | 528 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94.5 kB | 94.4 kB | N/A |
| page.js gzip | 151 kB | 150 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 621 B | 624 B | N/A |
| middleware-r..fest.js gzip | 151 B | 149 B | N/A |
| middleware.js gzip | 44.6 kB | 47.4 kB | ⚠️ +2.82 kB |
| edge-runtime..pack.js gzip | 1.94 kB | 1.94 kB | ✓ |
| Overall change | 46.5 kB | 49.3 kB | ⚠️ +2.82 kB |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sam/cli/commander | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 166 kB | 166 kB | N/A |
| app-page-exp..prod.js gzip | 95.9 kB | 95.4 kB | N/A |
| app-page-tur..prod.js gzip | 97.6 kB | 97.2 kB | N/A |
| app-page-tur..prod.js gzip | 92 kB | 91.6 kB | N/A |
| app-page.run...dev.js gzip | 136 kB | 136 kB | N/A |
| app-page.run..prod.js gzip | 90.6 kB | 90.2 kB | N/A |
| app-route-ex...dev.js gzip | 22 kB | 22 kB | N/A |
| app-route-ex..prod.js gzip | 14.9 kB | 14.9 kB | N/A |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | N/A |
| app-route-tu..prod.js gzip | 14.6 kB | 14.7 kB | N/A |
| app-route.ru...dev.js gzip | 21.7 kB | 21.7 kB | N/A |
| app-route.ru..prod.js gzip | 14.6 kB | 14.7 kB | N/A |
| pages-api-tu..prod.js gzip | 9.47 kB | 9.43 kB | N/A |
| pages-api.ru...dev.js gzip | 9.74 kB | 9.7 kB | N/A |
| pages-api.ru..prod.js gzip | 9.47 kB | 9.43 kB | N/A |
| pages-turbo...prod.js gzip | 22.1 kB | 22 kB | N/A |
| pages.runtim...dev.js gzip | 22.7 kB | 22.7 kB | N/A |
| pages.runtim..prod.js gzip | 22.1 kB | 22 kB | N/A |
| server.runti..prod.js gzip | 50.1 kB | 49.9 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>image-HASH.js</strong></summary>

```diff
@@ -1,7 +1,7 @@
 (self["webpackChunk_N_E"] = self["webpackChunk_N_E"] || []).push([
   [358],
   {
-    /***/ 1552: /***/ function (
+    /***/ 4070: /***/ function (
       __unused_webpack_module,
       __unused_webpack_exports,
       __webpack_require__
@@ -9,7 +9,7 @@
       (window.__NEXT_P = window.__NEXT_P || []).push([
         "/image",
         function () {
-          return __webpack_require__(2599);
+          return __webpack_require__(2369);
         },
       ]);
       if (false) {
@@ -18,7 +18,7 @@
       /***/
     },
 
-    /***/ 9272: /***/ function (module, exports, __webpack_require__) {
+    /***/ 3299: /***/ function (module, exports, __webpack_require__) {
       "use strict";
       /* __next_internal_client_entry_do_not_use__  cjs */
       Object.defineProperty(exports, "__esModule", {
@@ -40,15 +40,15 @@
         __webpack_require__(422)
       );
       const _head = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(8286)
+        __webpack_require__(4830)
       );
-      const _getimgprops = __webpack_require__(985);
-      const _imageconfig = __webpack_require__(8749);
-      const _imageconfigcontextsharedruntime = __webpack_require__(7829);
-      const _warnonce = __webpack_require__(3652);
-      const _routercontextsharedruntime = __webpack_require__(6120);
+      const _getimgprops = __webpack_require__(3301);
+      const _imageconfig = __webpack_require__(2958);
+      const _imageconfigcontextsharedruntime = __webpack_require__(6869);
+      const _warnonce = __webpack_require__(2179);
+      const _routercontextsharedruntime = __webpack_require__(7554);
       const _imageloader = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(5732)
+        __webpack_require__(9687)
       );
       // This is replaced by webpack define plugin
       const configEnv = {
@@ -69,8 +69,7 @@
         onLoadRef,
         onLoadingCompleteRef,
         setBlurComplete,
-        unoptimized,
-        sizesInput
+        unoptimized
       ) {
         const src = img == null ? void 0 : img.src;
         if (!img || img["data-loaded-src"] === src) {
@@ -166,7 +165,6 @@
             onLoadingCompleteRef,
             setBlurComplete,
             setShowAltText,
-            sizesInput,
             onLoad,
             onError,
             ...rest
@@ -221,8 +219,7 @@
                     onLoadRef,
                     onLoadingCompleteRef,
                     setBlurComplete,
-                    unoptimized,
-                    sizesInput
+                    unoptimized
                   );
                 }
               },
@@ -234,7 +231,6 @@
                 setBlurComplete,
                 onError,
                 unoptimized,
-                sizesInput,
                 forwardedRef,
               ]
             ),
@@ -246,8 +242,7 @@
                 onLoadRef,
                 onLoadingCompleteRef,
                 setBlurComplete,
-                unoptimized,
-                sizesInput
+                unoptimized
               );
             },
             onError: (event) => {
@@ -351,7 +346,6 @@
                 onLoadingCompleteRef: onLoadingCompleteRef,
                 setBlurComplete: setBlurComplete,
                 setShowAltText: setShowAltText,
-                sizesInput: props.sizes,
                 ref: forwardedRef,
               }),
               imgMeta.priority
@@ -379,7 +373,7 @@
       /***/
     },
 
-    /***/ 985: /***/ function (
+    /***/ 3301: /***/ function (
       __unused_webpack_module,
       exports,
       __webpack_require__
@@ -395,9 +389,9 @@
           return getImgProps;
         },
       });
-      const _warnonce = __webpack_require__(3652);
-      const _imageblursvg = __webpack_require__(1668);
-      const _imageconfig = __webpack_require__(8749);
+      const _warnonce = __webpack_require__(2179);
+      const _imageblursvg = __webpack_require__(892);
+      const _imageconfig = __webpack_require__(2958);
       const VALID_LOADING_VALUES =
         /* unused pure expression or super */ null && [
           "lazy",
@@ -766,7 +760,7 @@
       /***/
     },
 
-    /***/ 1668: /***/ function (__unused_webpack_module, exports) {
+    /***/ 892: /***/ function (__unused_webpack_module, exports) {
       "use strict";
       /**
        * A shared function, used on both client and server, to generate a SVG blur placeholder.
@@ -821,7 +815,7 @@
       /***/
     },
 
-    /***/ 9582: /***/ function (
+    /***/ 3612: /***/ function (
       __unused_webpack_module,
       exports,
       __webpack_require__
@@ -848,10 +842,10 @@
         },
       });
       const _interop_require_default = __webpack_require__(2430);
-      const _getimgprops = __webpack_require__(985);
-      const _imagecomponent = __webpack_require__(9272);
+      const _getimgprops = __webpack_require__(3301);
+      const _imagecomponent = __webpack_require__(3299);
       const _imageloader = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(5732)
+        __webpack_require__(9687)
       );
       function getImageProps(imgProps) {
         const { props } = (0, _getimgprops.getImgProps)(imgProps, {
@@ -883,7 +877,7 @@
       /***/
     },
 
-    /***/ 5732: /***/ function (__unused_webpack_module, exports) {
+    /***/ 9687: /***/ function (__unused_webpack_module, exports) {
       "use strict";
 
       Object.defineProperty(exports, "__esModule", {
@@ -918,7 +912,7 @@
       /***/
     },
 
-    /***/ 2599: /***/ function (
+    /***/ 2369: /***/ function (
       __unused_webpack_module,
       __webpack_exports__,
       __webpack_require__
@@ -939,8 +933,8 @@
 
       // EXTERNAL MODULE: ./node_modules/.pnpm/react@18.2.0/node_modules/react/jsx-runtime.js
       var jsx_runtime = __webpack_require__(1527);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@18.2.0_react@18.2.0/node_modules/next/image.js
-      var next_image = __webpack_require__(1577);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@18.2.0_react@18.2.0/node_modules/next/image.js
+      var next_image = __webpack_require__(73);
       var image_default = /*#__PURE__*/ __webpack_require__.n(next_image); // CONCATENATED MODULE: ./pages/nextjs.png
       /* harmony default export */ var nextjs = {
         src: "/_next/static/media/nextjs.cae0b805.png",
@@ -970,12 +964,12 @@
       /***/
     },
 
-    /***/ 1577: /***/ function (
+    /***/ 73: /***/ function (
       module,
       __unused_webpack_exports,
       __webpack_require__
     ) {
-      module.exports = __webpack_require__(9582);
+      module.exports = __webpack_require__(3612);
 
       /***/
     },
@@ -986,7 +980,7 @@
       return __webpack_require__((__webpack_require__.s = moduleId));
     };
     /******/ __webpack_require__.O(0, [888, 774, 179], function () {
-      return __webpack_exec__(1552);
+      return __webpack_exec__(4070);
     });
     /******/ var __webpack_exports__ = __webpack_require__.O();
     /******/ _N_E = __webpack_exports__;
```
</details>
<details>
<summary>Diff for <strong>68-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 4a0d8dc58f048f9d910b8f7cca7c643e0b68166d

File: packages/next/src/bin/next.ts [hunk 1]
```diff
@@ -68,75 +30,297 @@ if (
...
+  .addOption(
+    new Option('--build-mode [mode]')
+      .choices(['experimental-compile', 'experimental-generate'])
+      .default('default')
+      .hideHelp()
+  )
+  .action((directory, options) =>
+    // ensure process exits after build completes so open handles/connections
+    // don't cause process to hang
+    nextBuild(options, directory).then(process.exit(0))
```
Comment (ijjk) [inline packages/next/src/bin/next.ts:129]:
```suggestion
    nextBuild(options, directory).then(() => process.exit(0))
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -16,6 +16,11 @@
 /packages/create-next-app/                    @vercel/next-js
 /pnpm-lock.yaml                               @vercel/next-js @vercel/turbopack
 
+# Next.js CLI
+
+/packages/next/src/bin                           @samcx @ijjk
+/packages/next/src/cli                           @samcx @ijjk
+
```
Comment (ijjk) [inline .github/CODEOWNERS:23]: Wanna move this to a separate PR? Also maybe keep the same three that we have on all other specific owners: (`@timneutkens @ijjk @shuding`)

File: packages/next/package.json [hunk 3]
```diff
@@ -209,6 +209,7 @@
     "ci-info": "watson/ci-info#f43f6a1cefff47fb361c88cf4b943fdbcaafe540",
     "cli-select": "1.1.2",
     "client-only": "0.0.1",
+    "commander": "12.0.0",
```
Comment (ijjk) [inline packages/next/package.json:212]: Could we bump CNA's commander to to match here or are there breaking changes that warrant separate PR? 

x-ref: https://github.com/vercel/next.js/blob/8a2460ccb1d5dc480eb5d0f501daf8f839adb286/packages/create-next-app/package.json#L42

Comment (samcx) [reply packages/next/package.json:212]: Yeah I need to fix CNA as well. Separate <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>

File: test/integration/telemetry/test/index.test.js [hunk 4]
```diff
@@ -90,7 +90,7 @@ describe('Telemetry CLI', () => {
   it('can disable telemetry with env NEXT_TELEMETRY_DISABLED', async () => {
     // next config is not reset between tests
     await runNextCommand(['telemetry', 'enable'])
-    const { stdout } = await runNextCommand(['telemetry', 'status'], {
+    const { stdout } = await runNextCommand(['telemetry'], {
```
Comment (ijjk) [inline test/integration/telemetry/test/index.test.js:93]: This change shouldn't be necessary, let's ensure this continues to work to avoid breaking change

File: .github/CODEOWNERS (hunk 2 above)
Comment (samcx) [reply .github/CODEOWNERS]: <picture data-single-emoji=":done:" title=":done:">:done:</picture> , will create in separate <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>

Comment (samcx) [reply .github/CODEOWNERS]: https://github.com/vercel/next.js/pull/62749

File: packages/next/src/bin/next.ts [hunk 5]
```diff
@@ -68,75 +23,324 @@ if (
...
+  )
+  .option('-d, --debug', 'Enables a more verbose build output.')
+  .option('--profile', 'Enables production profiling for React.')
+  .option('--no-lint', 'Disables linting.')
+  .option('--no-mangling', 'Disables mangling.')
+  .addOption(
+    new Option(
+      '--experimental-app-only',
+      'Builds only App Router routes.'
+    ).hideHelp()
```
Comment (ijjk) [inline packages/next/src/bin/next.ts:115]: Let's keep this in the help output like it was before

File: packages/next/src/bin/next.ts [hunk 6]
```diff
@@ -68,75 +23,324 @@ if (
...
+  )
+  .addOption(new Option('--experimental-turbo').hideHelp())
+  .addOption(
+    new Option('--build-mode [mode]')
+      .choices(['experimental-compile', 'experimental-generate'])
+      .default('default')
+      .hideHelp()
```
Comment (ijjk) [inline packages/next/src/bin/next.ts:122]: Let's also ensure this is in the help output

File: packages/next/src/bin/next.ts [hunk 7]
```diff
@@ -68,75 +23,324 @@ if (
...
+  .option('--no-mangling', 'Disables mangling.')
+  .addOption(
+    new Option(
+      '--experimental-app-only',
+      'Builds only App Router routes.'
+    ).hideHelp()
+  )
+  .addOption(new Option('--experimental-turbo').hideHelp())
+  .addOption(
+    new Option('--build-mode [mode]')
```
Comment (ijjk) [inline packages/next/src/bin/next.ts:119]:
```suggestion
    new Option('--experimental-build-mode [mode]')
```

File: packages/next/src/bin/next.ts [hunk 8]
```diff
@@ -68,75 +23,324 @@ if (
...
+      .choices(['experimental-compile', 'experimental-generate'])
```
Comment (ijjk) [inline packages/next/src/bin/next.ts:120]:
```suggestion
      .choices(['compile', 'generate'])
```

File: packages/next/src/bin/next.ts (hunk 5 above)
Comment (samcx) [reply packages/next/src/bin/next.ts]: <picture data-single-emoji=":done:" title=":done:">:done:</picture>

**Review by ijjk — APPROVED** [review]
Good work, looks good to me!
